Source/SwiftLintFramework/Rules/Performance/ReduceIntoRule.swift

```diff
@@ -0,0 +1,155 @@
+import Foundation
+import SourceKittenFramework
+
+public struct ReduceIntoRule: ASTRule, ConfigurationProviderRule, OptInRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static var description = RuleDescription(
+        identifier: "reduce_into",
+        name: "Reduce Into",
+        description: "Prefer `reduce(into:_:)` over `reduce(_:_:)` for copy-on-write types",
+        kind: .performance,
+        minSwiftVersion: .three,
```

Comment:
Thanks for the great contribution, @daltonclaybrook!

`reduce(into:)` was added in Swift 4 however, not Swift 3: https://github.com/apple/swift-evolution/blob/master/proposals/0171-reduce-with-inout.md

Reply:
Bumped it to 4 in 230bedcd.